### PR TITLE
RHBRMS-2744: Persistence descriptor Validation button does nothing

### PR DESCRIPTION
**Pull request by wmedvede**
- adds the missing validation of the persistence descriptor

### CODE REVIEW

**Comment by wmedvede** [comment]
Hello @pefernan this PR implements the missing piece for the persistence descriptor editor we talked.
Would you mind review?

Thanks

**Comment by pefernan** [comment]
jenkins retest this.

**Comment by pefernan** [comment]
@wmedvede looks good, +1 when green

**Comment by wmedvede** [comment]
Guys, @porcelli @manstis i'll appreciate strongly if this can be merged for the 7.0 community.
Thanks!

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/validation/PersistenceDescriptorValidatorImpl.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            messages.add( newErrorMessage( PersistenceDescriptorValidationMessages.PERSISTENCE_UNIT_NAME_EMPTY_ID,
+                    PersistenceDescriptorValidationMessages.PERSISTENCE_UNIT_NAME_EMPTY ) );
+        }
+
+        if ( unitModel.getProvider( ) == null || unitModel.getProvider( ).trim( ).isEmpty( ) ) {
+            messages.add( newErrorMessage( PersistenceDescriptorValidationMessages.PERSISTENCE_UNIT_PROVIDER_ID,
+                    PersistenceDescriptorValidationMessages.PERSISTENCE_UNIT_PROVIDER_EMPTY ) );
+        }
+
+        if ( unitModel.getTransactionType( ) == null ) {
```
Comment (jsoltes) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/validation/PersistenceDescriptorValidatorImpl.java]: Maybe here could be used switch-case statement instead...

**Comment by wmedvede** [comment]
@porcelli the modification is in place.
Thanks

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/validation/PersistenceDescriptorValidatorImpl.java (hunk 1 above)
Comment (porcelli) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/validation/PersistenceDescriptorValidatorImpl.java]: not possible to use switch-case statement for null cases, so basically would need to create a data structure to execute the conditions... can't see much of the benefit

**Pull request merged**